subsys/net/ip/net_context.c

```diff
@@ -825,10 +825,19 @@ NET_CONN_CB(tcp_established)
 	}
 
 	set_appdata_values(pkt, IPPROTO_TCP);
-	context->tcp->send_ack += net_pkt_appdatalen(pkt);
+
+	uint16_t data_len = net_pkt_appdatalen(pkt);
+	if (data_len > get_recv_wnd(context->tcp)) {
+		NET_ERR("Context %p: overflow of recv window (%d vs %d), pkt dropped",
+			context, get_recv_wnd(context->tcp), data_len);
+		return NET_DROP;
```

Comment:
Well, that's a dream scenario of any DoS attacker - they send you useless stuff which you try to hold, while dropping real precious things, because all your hands are tied with useless stuff.